pipeline/installs/multi-user/pipelines-profile-controller/sync.py

```diff
@@ -0,0 +1,274 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from http.server import BaseHTTPRequestHandler, HTTPServer
+import json
+import os
+
+kfp_version = os.environ["KFP_VERSION"]
+disable_istio_sidecar = os.environ.get("DISABLE_ISTIO_SIDECAR") == "true"
+
+
+class Controller(BaseHTTPRequestHandler):
+    def sync(self, parent, children):
+        # HACK: Currently using serving.kubeflow.org/inferenceservice to identify
+        # kubeflow user namespaces.
+        # TODO: let Kubeflow profile controller add a pipeline specific label to
+        # user namespaces and use that label instead.
+        pipeline_enabled = parent.get("metadata", {}).get(
+            "labels", {}).get("serving.kubeflow.org/inferenceservice")
+
+        if not pipeline_enabled:
+            return {"status": {}, "children": []}
+
+        # Compute status based on observed state.
+        desired_status = {
+            "kubeflow-pipelines-ready": \
+                len(children["Secret.v1"]) == 1 and \
+                len(children["ConfigMap.v1"]) == 1 and \
+                len(children["Deployment.apps/v1"]) == 2 and \
+                len(children["Service.v1"]) == 2 and \
+                len(children["DestinationRule.networking.istio.io/v1alpha3"]) == 1 and \
+                len(children["ServiceRole.rbac.istio.io/v1alpha1"]) == 1 and \
+                len(children["ServiceRoleBinding.rbac.istio.io/v1alpha1"]) == 1 and \
+                "True" or "False"
+        }
+
+        # Generate the desired child object(s).
+        # parent is a namespace
+        namespace = parent.get("metadata", {}).get("name")
+        desired_resources = [
+            {
+                "apiVersion": "v1",
+                "kind": "Secret",
+                "metadata": {
+                    "name": "mlpipeline-minio-artifact",
+                    "namespace": namespace,
+                },
+                "data": {
+                    "accesskey": "bWluaW8=",  # base64 for minio
```

Comment:
Are we really checking in access keys? Is this safe?

Reply:
Yes, it would be recommended to customize it after deployment.